ros2doctor/ros2doctor/api/format.py

```diff
@@ -12,9 +12,33 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import sys
 
-def print_term(k, v):
-    """Print term only if it exists."""
-    if v:
-        # TODO(clairewang): 20 padding needs to be dynamically set
-        print('{:20}: {}'.format(k, v))
+
+def format_print(modules, report):
+    """Format print report according to module names."""
+    padding_num = compute_padding(modules, report)
+    for m in modules:
+        if m in report:
+            module_report = report.get(m)
+            for k, v in module_report:
+                if k == 'NAME':
```

Comment:
Recommend having reports return a class instance (`class Report:`?) instead of a dictionary. It's hard to document that `NAME` is a special key reports can return; where does that documentation go? A class definition with docstrings would be a spot that can be pointed to when answering future questions about how to write a report for `ros2doctor`.